examples/jsm/controls/TransformControls.js

```diff
@@ -215,7 +215,13 @@ class TransformControls extends Object3D {
 
 		}
 
-		super.updateMatrixWorld( force );
+		// TransformControls should ignore any parent matrix
+		this.matrixAutoUpdate = false;
+		this.updateMatrix();
+		this.matrixWorldNeedsUpdate = false;
+		this.matrixWorld.copy( this.matrix );
+
+		super.updateMatrixWorld( false );
```

Comment:
I can confirm your PR fixes the issue but the approach isn't ideal, tbh. We need to find a solution _without_ influencing how the world transformation matrix is computed. 

If not, this will cause issues when we rethink the update matrix process and lifecycle in `three.js` and try to find a more optimized implementation. Ideally, components just request a world matrix and do not modify their computation.

In other words, `TransformControls` has to work with an unmodified version of the scene's hierarchy. Fixing the controls in combination with rotated ancestor nodes has to be done in a way that the rotation is correctly taken into account and not ignored.

Reply:
Would this match your requirements ? 

```
// if TransformControls has a parent, set its local position to the inverse of the parent in order to have a worldMatrix to identity
if ( this.parent ) {

			this._transformMatrixWorldInv.copy( this.parent.matrixWorld ).invert();
			this._transformMatrixWorldInv.decompose( this.position, this.rotation, this.scale );

}

super.updateMatrixWorld( true );
```

That way we don't influence how the world transformation matrix is computed, we just set the TransformControls local position.

Reply:
That looks indeed better.

Reply:
Awesome! 
I'll push that commit later today

Reply:
updated, it should be good now 👍